sdk/core/System.Net.ClientModel/tests/client/MapsClientTests.cs

```diff
@@ -26,4 +28,52 @@ public void TestClientSync()
         Assert.AreEqual("US", result.Value.CountryRegion.IsoCode);
         Assert.AreEqual(IPAddress.Parse("2001:4898:80e8:b::189"), result.Value.IpAddress);
     }
+
+    [Test]
+    public void TestCustomPipelineForMethodInvocation()
+    {
+        string key = Environment.GetEnvironmentVariable("MAPS_API_KEY");
+
+        KeyCredential credential = new KeyCredential(key);
+
+        // We have to pass the pipeline options to the client constructor
+        // so that the auth policy gets added, otherwise we won't have it later.
+        PipelineOptions pipelineOptions = new PipelineOptions();
+
+        MapsClient client = new MapsClient(new Uri("https://atlas.microsoft.com"), credential, pipelineOptions);
+
+        // Add a custom policy to the pipeline just for the one method
+        pipelineOptions.PerCallPolicies = new PipelinePolicy[1];
```

Comment:
Left some comments about this here: https://github.com/Azure/azure-sdk-for-net/pull/38428#discussion_r1385296589

Reply:
We reworked this to add a convenience method to RequestOptions for adding policies to make that easy, and the client stores client-specific policies that it passes to MessagePipeline.GetPipeline when it freezes the pipeline on options.